docs/developing-ray-libraries/developing-a-ray-library.md

```diff
@@ -0,0 +1,411 @@
+
+Here's what is being sent from our library to Ray, according to the proxy:
+
+```json
+{
+  "uuid": "8bd2e386-e000-47b6-bd31-a655fd66376d",
+  "payloads": [
+    {
+      "type": "log",
+      "content": {
+        "content": "log",
+        "label": "log"
+      },
+      "origin": {
+        "function_name": "my_test_func",
+        "file": "my-file.js",
+        "line_number": 16,
+        "hostname": "my-hostname"
+      }
+    }
+  ],
+  "meta": {
+    "my_package_version": "1.0.0"
+  }
+}
+```
+
+Run the following, to see what we should be sending:
+
+```bash
+php ./ray-library-reference/ray-test.php
+```
+
+And here's what SHOULD be sent _(relevant parts only)_:
+
+```json
+{
+  ...
+  "payloads": [
+    {
+      "type": "log",
+      "content": {
+        "values": [
+          "Hello World!"
+        ]
+      },
+	...
+    }
+  ],
+  "meta": {
+	...
+  }
+}
+```
+
+The issue appears to be with our `createLogPayload` function, so let's change a few things:
+
+```typescript
+export function createLogPayload(text: string|string[], uuid: string | null = null) {
+	// make sure we pass an array of values
+    if (!Array.isArray(text)) {
+        text = [text];
+    }
+    // add the last parameter to send "content.values" in the payload
+	const payload = createPayload('log', 'log', text, 'values');
+	return createSendablePayload([payload], uuid);
+}
+```
+
+And lastly, we need to update the `send()` method on the `Ray` class:
+
+```typescript
+public send(...args: any[]): Ray {
+    // we only need to send a single payload with `args` as the data, instead of
+    // sending a new payload for each arg in args.
+	const payload = createLogPayload(args, this.uuid);			
+	this.sendRequest(payload);
+	
+	return this;
+}
+```
+
+Let's compile again, and re-run our test script:
+
+```bash
+./node_modules/.bin/esbuild --bundle \
+  --target=node12 --platform=node \
+  --format=cjs --external:superagent \
+  --outfile=dist/index.js src/Ray.ts
+
+node dist/test.js
+```
+
+Success!  You should see two "hello world" messages, one red and one blue.
+
+Let's make one more change to make using our library easier: adding a `ray()` function to `./src/Ray.ts`:
+
+```typescript
+// ... Ray class code here
+
+export function ray(...args: any[]) {
+    return (new Ray()).send(...args);
+}
+
+export default Ray;
+```
+
+You can now modify your test script to something like:
+
+```typescript
+const { ray } = require('./index');
+
+ray('hello world').color('red');
+ray().html('<strong>bold text</strong>');
+```
+
+That's it! Make sure to take a look at the [companion repository](https://github.com/permafrost-dev/creating-a-ray-integration) to check out the final project code.
+
+> Don't forget to stop the `ray-proxy` app and change your Ray port back to 23517.
```

Comment:
Is this `">"` character possibly breaking the HTML?

Reply:
I'm thinking it shouldn't be a problem, but let's remove it. Better safe than sorry.